packages/apputils/src/dialog.tsx

```diff
@@ -436,12 +436,14 @@ export class Dialog<T> extends Widget {
         event.preventDefault();
 
         const activeEl = document.activeElement;
-        let index: number | undefined;
+        let index: number | undefined = this._defaultButton;
 
         if (activeEl instanceof HTMLButtonElement) {
           index = this._buttonNodes.indexOf(activeEl);
```

Comment:
Index here is always overwritten, we may as well remove the assignment above to reduce cognitive complexity

```suggestion
          const index = this._buttonNodes.indexOf(activeEl);
```